### ISSUE TITLE
At most two tracebacks displayed in case of cascading errors

### ISSUE THREAD

**Issue by proto3** [agronholm/exceptiongroup#36]
I have a problem since the 1.0.0 release. Only the two first tracebacks are displayed in case of cascading errors.
I'm actually not using exceptiongroup features, but since pytest is using it, it changes the normal tracebacks behavior in my tests.

**EXPECTED:**
Here is a code with 3 nested functions producing cascading errors:
```python
#!/usr/bin/env python

def func3():
    raise Exception("LEVEL_3")

def func2():
    try:
        func3()
    except Exception:
        raise Exception("LEVEL_2")

def func1():
    try:
        func2()
    except Exception:
        raise Exception("LEVEL_1")

if __name__ == "__main__":
    func1()
```
And here is the output I get:
```
Traceback (most recent call last):
  File "/home/felix/projects/bliss/./repro_bug.py", line 9, in func2
    func3()
  File "/home/felix/projects/bliss/./repro_bug.py", line 5, in func3
    raise Exception("LEVEL_3")
Exception: LEVEL_3

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/felix/projects/bliss/./repro_bug.py", line 15, in func1
    func2()
  File "/home/felix/projects/bliss/./repro_bug.py", line 11, in func2
    raise Exception("LEVEL_2")
Exception: LEVEL_2

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/felix/projects/bliss/./repro_bug.py", line 20, in <module>
    func1()
  File "/home/felix/projects/bliss/./repro_bug.py", line 17, in func1
    raise Exception("LEVEL_1")
Exception: LEVEL_1
```

**PROBLEM:**
Now if I `import exceptiongroup` at the beginning to monkeypatch tracebacks, the last level is lost.
I can add more error levels, but I only see LEVEL_1 and LEVEL_2 at the end.
```
Traceback (most recent call last):
  File "/home/felix/projects/bliss/./repro_bug.py", line 21, in func1
    func2()
  File "/home/felix/projects/bliss/./repro_bug.py", line 17, in func2
    raise Exception("LEVEL_2")
Exception: LEVEL_2

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/felix/projects/bliss/./repro_bug.py", line 26, in <module>
    func1()
  File "/home/felix/projects/bliss/./repro_bug.py", line 23, in func1
    raise Exception("LEVEL_1")
Exception: LEVEL_1
```

**Comment by proto3** [comment]
After digging a bit, I think it is related to  [PatchedTracebackException.\__init__()](https://github.com/agronholm/exceptiongroup/blob/1.0.0/src/exceptiongroup/_formatting.py#L75)

There are multiple recursive calls inside the function:
- [traceback_exception_original_init(](https://github.com/agronholm/exceptiongroup/blob/1.0.0/src/exceptiongroup/_formatting.py#L91) line 91
- [cause = PatchedTracebackException(](https://github.com/agronholm/exceptiongroup/blob/1.0.0/src/exceptiongroup/_formatting.py#L116) line 116
- [context = PatchedTracebackException(](https://github.com/agronholm/exceptiongroup/blob/1.0.0/src/exceptiongroup/_formatting.py#L140) line 140
- [texc = PatchedTracebackException(](https://github.com/agronholm/exceptiongroup/blob/1.0.0/src/exceptiongroup/_formatting.py#L157) line 157

There is `is_recursive_call` boolean to prevent recursive calls from the line 91 to enter `if not is_recursive_call:`, but then all the three others are also affected and this stops the chaining of tracebacks prematurely.

**Comment by agronholm** [comment]
Do you know how to fix this?

**Comment by agronholm** [comment]
None of what you said actually explain the results. The upstream implementation has this variable too, along with recursive calls.

**Comment by agronholm** [comment]
I honestly don't see how moving that block of code would have any effect. I'm working on a modification that copies the entire py3.10 `TracebackException.__init__()` code to the patched version to avoid problems like this, but so far I've not been able to make it work for Pythons older than that. To make matters worse, I'm currently trying to resolve a semi-urgent issue in the `wheel` project.

**Comment by agronholm** [comment]
Ok, I have a PR that fixes the problem. Would you mind reviewing it?

**Comment by agronholm** [comment]
I've just released v1.0.1.
